usagov/usagov-2021
# Usagov 1697 category language

## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-1697

## Description

Adds a prerender hook that cleans up the views results and removes terms that don't match the requested language
Requires a cache rebuild for Drupal to pick up the new hook.

## Type of Changes

  - Add screenshot showing what it should look like

## Testing Instructions


Setup the benefits search pages and tag sample data, if you haven't already.

```
bin/drush php:script scripts/drush/benefits-category-tag-pages.php /var/www/scripts/drush/data/benefits-sample.2024-05-22.json
bin/drush php:script scripts/drush/benefits-category-make-pages.php
```

Edit one or more basic pages that are standard pages. In the benefit category checkboxes, select values from both English and Spanish.

### Requires New Config

### Requires New Content

### Validation Steps
- Edit at least one English and Spanish basic pages that are standard pages. In the benefit category check boxes, select values from both English and Spanish.
- In the English search results, pick categories that match the ones selected for the page you edited. Confirm only the English taxonomy terms are displayed alongside it.
- In the Spanish search results, pick categories that match the ones selected for the page you edited. Confirm only the Spanish taxonomy terms are displayed alongside it.

## Security Review




## Reviewer Reminders
- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions
Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
